tensorflow/compiler/xla/service/gpu/BUILD

```diff
@@ -1006,25 +1006,116 @@ cc_library(
         "//tensorflow/core:stream_executor_no_cuda",
         "@com_google_absl//absl/memory",
         "@llvm//:core",
-    ],
+    ] + if_cuda_is_configured([
```

Comment:
My personal preference is to combine all three targets (`gpu_compiler`, `nvptx_compiler` and `amdgpu_compiler`) into one (`gpu_compiler`). My reasoning is as follows:

- The bazel logic make the dependency largely the same between the three, especially between `nvptx_compiler` and `amdgpu_compiler`.
- Since this is all configure time, only either `gpu_compiler` + `nvptx_compiler` exists or `gpu_compiler` + `amdgpu_compiler` exists. Making them more compact reflect more closely with compilation process. We can have the `gpu_compiler.*` files as not conditioned, and platform dependent files guarded by if_cuda/rocm_is_configured.

Reply:
sounds like a good idea to me. let me do that.